### DISCUSSION TITLE
getLocation() Crashing, invalid parameter type

### DISCUSSION THREAD

**Discussion by varun-dubagunta-clubs** [SimDynamX/SpaceTeamsPRO#51] [Q&A] [answered]
Hello, I was running some test code to isolate my errors and the logs returned the following issue

TypeError: getLocation(): incompatible function arguments. The following argument types are supported:
    1. (self: spaceteams.SC_Compute_Server.Entity) -> spaceteams.SC_Compute_Server.FramedLoc

for reference, I have included the variable declaration that got passed in here:
moon1: sc.Entity  = this.GetParam(sc.VarType.entityRef, "moonFrame")
rpos = Rover1.getLocation(moonFrame)

**Comment by connorjak** [comment, marked as answer]
getLocation() no longer has any input args; we haven't updated our general Python API docs in a while. In VS Code, you can usually control-click the one of the `spaceteams`/`st` functions and that will take you to `SC_Compute_Server.pyi`, which is guaranteed to be up to date with all of the available function signatures.

For the purposes of the competition, I recommend using EntityTelemetry functions such as `ET.GetCurrentXY(entity)`

**Reply by varun-dubagunta-clubs** [reply]
Thank you for the reply! Quick follow up question: does the entity parameter refer to the frame?

**Reply by connorjak** [reply]
Entities have body-fixed frames, you can get a body-fixed frame for an entity with `.getBodyFixedFrame()`. Body-centered frame is also available (uses entity's resident frame for the attitude and the entity's location for the origin)

**Comment by connorjak** [comment]
Also be careful, we haven't put out the best information on this (will address it during Task 1 feedback); if you do things such as directly call getLocation() for the movable entities in the MissionManager script, that works around the loss-of-comms functionality and is invalid for the competition.
